src/sentry/auth/helper.py

```diff
@@ -171,7 +189,7 @@ def handle_existing_identity(
             tags={
                 "provider": self.provider.key,
                 "organization_id": self.organization.id,
-                "user_id": self.user.id,
+                "user_id": user.id,
```

Comment:
This is the only place that `handle_existing_identity` referred to `self.user`, and I'm pretty sure it was a mistake. It should have been using the local `user` variable from line 178 all along.

With this `self.user` gone, only `handle_unknown_identity` and `handle_attach_identity` will trigger the warning from when `self.user` is lazily initialized. Although `handle_unknown_identity` dispatches to `handle_attach_identity`, the warning will be generated only once per `AuthIdentityHandler` instance, which is nice. (It still may be generated multiple times per login flow as we respond to several HTTP requests in a row, but there's no simple way to prevent that.)

As a bonus, this will get rid of some unnecessary DB lookups when only `handle_existing_identity` is called.